src/components/common/ReferalLinkActivityIndicator.js

```diff
@@ -0,0 +1,127 @@
+// @flow
+
+import React, { Component, Fragment } from 'react'
+import { ActivityIndicator, Animated, StyleSheet, Text, View } from 'react-native'
+import { type AirshipBridge } from 'react-native-airship'
+
+import { THEME } from '../../theme/variables/airbitz.js'
+import { scale } from '../../util/scaling.js'
+import { Airship } from '../services/AirshipInstance.js'
+
+export function showActivity<T> (message: string, promise: Promise<T>): Promise<T> {
+  Airship.show(bridge => <AirshipToast bridge={bridge} message={message} activity={promise} />)
+  return promise
+}
+
+const fadeInTime = 300
+const fadeOutTime = 1000
+const visibleTime = 3000
+
+type Props = {
+  bridge: AirshipBridge<void>,
+  // The message to show in the toast:
+  message?: string,
+  // If set, the toast will stay up for the lifetime of the promise,
+  // and will include a spinner.
+  activity?: Promise<mixed>
+}
+
+class AirshipToast extends Component<Props> {
```

Comment:
We need to come up with a better name for this. `FullScreenActivity`?

Reply:
It is not the official, the indicator as of now, maybe I can just rename it as the filename?

Reply:
Yes, the file name & component should both be called `FullScreenSpinner` or something, and then move the `showFullScreenSpinner` function into services/AirshipInstance.js.